### PR TITLE
Vault updates need to happen after the client has been built.

### PR DESCRIPTION
**Pull request by mcole**
Specifying --bootstrap-vault-item etc. on bootstrap to encrypt keys for a new node did not work because chef-vault was executed before the client has been created.

### CODE REVIEW

**Comment by chef-supermarket** [comment]
Hi. I am an automated pull request bot named Curry. There are commits in this pull request whose authors are not yet authorized to contribute to Chef Software, Inc. projects or are using a non-GitHub verified email address. To become authorized to contribute, you will need to sign the Contributor License Agreement (CLA) as an individual or on behalf of your company. [You can read more on Chef's blog.](https://www.chef.io/blog/2014/06/23/changes-to-the-contributor-license-agreement-process)

## Non-GitHub Verified Committers

There are 1 commit author(s) whose commits are authored by a non-GitHub verified email address. Chef will have to manually verify that they are authorized to contribute.

[Please sign the CLA here.](https://supermarket.chef.io/become-a-contributor)

File: lib/chef/knife/bootstrap_windows_base.rb [hunk 1]
```diff
@@ -300,10 +300,11 @@ def bootstrap(proto=nil)
             exit 1
           end
 
-          chef_vault_handler.run(node_name: config[:chef_node_name]) if chef_vault_handler.doing_chef_vault?
-
           client_builder.run
           bootstrap_context.client_pem = client_builder.client_path
+
+          chef_vault_handler.run(node_name: config[:chef_node_name]) if chef_vault_handler.doing_chef_vault?
```
Comment (mwrock) [inline lib/chef/knife/bootstrap_windows_base.rb]: Just to keep parity with chef core, we should put this between `client_builder.run` and `bootstrap_context.client_pem = client_builder.client_path`.

I don't think there are any functional ramifications there.

**Comment by mwrock** [comment]
I'm generally :+1:  on this with my comment above.

**Comment by mwrock** [comment]
cc @chef/client-windows

**Comment by jaym** [comment]
test failures look unrelated :+1:

**Comment by mwrock** [comment]
Thanks @mcole !!

**Pull request merged**